runtime/doc/lsp.txt

```diff
@@ -700,10 +701,30 @@ Lua module: vim.lsp                                                 *lsp-core*
       • {filetypes}?     (`string[]`) Filetypes the client will attach to, if
                          activated by `vim.lsp.enable()`. If not provided,
                          then the client will attach to all filetypes.
-      • {root_markers}?  (`string[]`) Directory markers (.e.g. '.git/') where
-                         the LSP server will base its workspaceFolders,
-                         rootUri, and rootPath on initialization. Unused if
-                         `root_dir` is provided.
+      • {root_markers}?  (`(string|string[])[]`) Directory markers (.e.g.
+                         '.git/') where the LSP server will base its
+                         workspaceFolders, rootUri, and rootPath on
+                         initialization. Unused if `root_dir` is provided.
+
+                         It's possibile to specify marker's priority. The
+                         default behaviour performs a directory-up first
+                         search, but it can be configured to look for multple
+                         markers in each directory before going up.
+
+                         Example (default, directory-up first): >lua
+                               root_markers = { 'stylua.toml', '.git' }
+<
+
+                         Looks for `stylua.toml` in every directory going up,
+                         if it isn't found it does the same for `.git`.
+
+                         Example (mixed priorities): >lua
+                               root_markers = { { 'stylua.toml', '.luarc.json' }, '.git' }
```

Comment:
should we require the "nested" case to have all of its items as lists? else it complicates the logic.
```suggestion
                               root_markers = { { 'stylua.toml', '.luarc.json' }, { '.git' } }
```

Reply:
Afaik it does not complicate the logic at all. We simply iterate root_markers and pass each elem (`string` or `string[]`) directly to `vim.fs.root`.

Reply:
This is actually something that @lewis6991 suggested. https://github.com/neovim/neovim/pull/33485#discussion_r2055671928

Reply:
Yeah true, since the `else` was [eliminated](https://github.com/neovim/neovim/pull/33485#discussion_r2056954917).